pinnable.go

```diff
@@ -0,0 +1,38 @@
+package ebpf
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/cilium/ebpf/internal"
+)
+
+func pin(currentPath, newPath string, fd *internal.FD) error {
+	if newPath == "" {
+		return fmt.Errorf("pinned path cannot be empty")
+	}
+	if currentPath != "" {
```

Comment:
I would remove this condition and indentation block in favor of an early return with `currentPath == newPath { return nil }`, see other comments. I try to keep the happy path on the left as it tends to improve readability.